experimental/QuadFormAndIsom/src/enumeration.jl

```diff
@@ -787,144 +1199,205 @@ julia> splitting_of_prime_power(Lf, 3, 1)
-                                                                n_inv::Int=-1,
-                                                                pB::Int=-1,
-                                                                nB::Int=-1,
-                                                                fix_root::Bool=false)
+function splitting_of_pure_mixed_prime_power(
+    Lf::ZZLatWithIsom,
+    p::Int;
+    eiglat_cond::Dict{Int, Vector{Int}} = Dict{Int, Vector{Int}}(),
+    fix_root::Int=-1,
+    genusDB::Union{Nothing, Dict{ZZGenus, Vector{ZZLat}}}=nothing,
+    root_test::Bool=false,
+    info_depth::Int=1,
+  )
   rank(Lf) == 0 && return ZZLatWithIsom[Lf]
 
-  @req iseven(Lf) "Lattice must be even"
-
   n = order_of_isometry(Lf)
 
+  @req iseven(Lf) "Lattice must be even"
   @req is_finite(n) "Isometry must be of finite order"
-
-  p = typeof(n)(_p)
   @req is_prime(p) "p must be a prime number"
 
   pd = prime_divisors(n)
 
-  @req 1 <= length(pd) <= 2 && p in pd "Order must be divisible by p and have at most 2 prime divisors"
+  @req length(pd) <= 2 && p in pd "Order must be divisible by p and have at most 2 prime divisors"
 
-  if length(pd) == 2
-    q = pd[1] == p ? pd[2] : pd[1]
-    d = valuation(n, p)::Int  # Weird ? Valuation is type stable and returns Int but it bugs here
-    e = valuation(n, q)::Int
-  else
-    q = one(n)
-    d = valuation(n, p)::Int
-    e = zero(n)
+  # In that case (L, f) is of hermitian type, so we can call the appropriate
+  # function
+  if length(pd) == 1
+    return representatives_of_hermitian_type(Lf, p, fix_root; cond=get(eiglat_cond, p*n, Int[-1, -1, -1]), genusDB, root_test, info_depth)
   end
 
+  q = pd[1] == p ? pd[2] : pd[1]
+  d = valuation(n, p)::Int  # Weird ? Valuation is type stable and returns Int but it bugs here
+  e = valuation(n, q)::Int
+
   phi = minimal_polynomial(Lf)
   chi = prod(cyclotomic_polynomial(p^d*q^i, parent(phi)) for i in 0:e; init=one(phi))
 
   @req is_divisible_by(chi, phi) "Minimal polynomial is not of the correct form"
 
   reps = ZZLatWithIsom[]
 
-  if e == 0
-    return representatives_of_hermitian_type(Lf, p, fix_root)
-  end
-
-  A0 = kernel_lattice(Lf, p^d*q^e)
-  if pB >= 0 && signature_tuple(A0)[1] != pB
-    return reps
-  elseif nB >= 0 && signature_tuple(A0)[3] != nB
-    return reps
-  end
-  bool, r = divides(phi, cyclotomic_polynomial(p^d*q^e, parent(phi)))
+  bool, r = divides(phi, cyclotomic_polynomial(n, parent(phi)))
   @hassert :ZZLatWithIsom 1 bool
 
-  B0 = kernel_lattice(Lf, r)
-  RA = representatives_of_hermitian_type(A0, p, fix_root)
-  is_empty(RA) && return reps
-  RB = splitting_of_pure_mixed_prime_power(B0, p; p_inv, n_inv)
+  # We follow Algorithm 6 of [BH23]: there is a slight mistake in the pseudocode
+  # though, `A_0` and `B_0` should be switched for calling the algorithm
+  # `PrimitiveExtensions`.
+  A0 = kernel_lattice(Lf, r)
+  B0 = kernel_lattice(Lf, n)
```

Comment:
Funny small improvement: the function decomposes `n` as $p^e*q^d$ and later we redo this product instead of using $n$ itself...